packages/alea-frontend/pages/api/university-admin/semester-courses/create-new-course.ts

```diff
@@ -0,0 +1,78 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { Action, ResourceName } from '@alea/utils';
+import { getUserIdIfAuthorizedOrSetError } from '../../access-control/resource-utils';
+import { checkIfPostOrSetError, executeAndEndSet500OnError } from '../../comment-utils';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!checkIfPostOrSetError(req, res)) return;
+
+  const { universityId, instanceId, courseId } = req.body;
+
+  if (!universityId || !instanceId || !courseId) {
+    return res.status(422).end('Missing required fields: universityId, instanceId, courseId');
+  }
+
+  const userId = await getUserIdIfAuthorizedOrSetError(
+    req,
+    res,
+    ResourceName.UNIVERSITY_SEMESTER_DATA,
+    Action.MUTATE,
+    { universityId }
+  );
+  if (!userId) return;
+
+  const existing = await executeAndEndSet500OnError(
+    `SELECT 1 FROM courseMetadata WHERE courseId = ? AND instanceId = ?`,
+    [courseId, instanceId],
+    res
+  );
+
+  if (!existing) return;
+
+  if (Array.isArray(existing) && existing.length > 0) {
+    return res.status(200).json({ message: 'Course already exists for this semester' });
+  }
+
+  const insertResult = await executeAndEndSet500OnError(
+    `INSERT INTO courseMetadata (
+      courseId,
+      instanceId,
+      universityId,
+      lectureSchedule,
+      tutorialSchedule,
+      hasHomework,
+      hasQuiz,
+      updaterId,
+      courseName,
+      notes,
+      landing,
+      slides,
+      institution,
+      teaser,
+      instructors
+    ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`,
+    [
+      courseId,
+      instanceId,
+      universityId,
+      JSON.stringify([]),
+      JSON.stringify([]),
+      false,
+      false,
+      userId,
+      courseId,
+      '',
+      '',
+      '',
+      null,
+      null,
+      JSON.stringify([]),
+    ],
+    res
+  );
+
+  if (!insertResult) return;
+
+  res.status(201).json({ message: 'New course created successfully' });
```

Comment:
Please cleanuop these unnecessary json returns
```suggestion
  res.status(201).end();
```